File: vgjewellry/vg_jewellery/page/vg_quotation_cart/vg_quotation_cart.js
frappe.pages['vg-quotation-cart'].on_page_load = function(wrapper) {

	new QuotationCart(wrapper);

};

class QuotationCart {

	constructor(wrapper){

		this.wrapper = $(wrapper);

		this.page = frappe.ui.make_app_page({
			parent: wrapper,
			title: "Quotation Cart",
			single_column: true
		});

		this.make();

		this.load_suppliers();

		this.bind_events();

	}

	make(){

		this.page.main.html(`

	    <div class="quotation-cart-page">

		<div class="supplier-table"></div>

	    </div>

	`);
		this.page.add_inner_button(
			'<i class="fa fa-arrow-left"></i> Quotation Bank',
			() => {
				frappe.set_route("quotation-bank");
			})
		this.page.add_inner_button(
			'<i class="fa fa-arrow-right"></i> Quotation PO',
			() => {
				frappe.set_route("vg-quotation-po");
			}
		);

	}

	load_suppliers(){

		frappe.call({

			method: "vgjewellry.vg_jewellery.page.vg_quotation_cart.vg_quotation_cart.get_supplier_list",

			callback: (r)=>{

				this.render_table(r.message || []);

			}

		});

	}

	render_table(data){

		let html = `

	<table class="table table-bordered table-hover">

	    <thead>

		<tr>

		    <th width="180">Supplier Code</th>

		    <th>Supplier Name</th>

		    <th width="120">Jewellery</th>

		    <th width="100">Qty</th>

		    <th width="120">Action</th>

		</tr>

	    </thead>

	    <tbody>

	`;

		data.forEach(d=>{

			html += `

		<tr>

		    <td>${d.vendor_code || ""}</td>

		    <td>${d.vendor}</td>

		    <td>Diamond</td>

		    <td>${d.qty}</td>

		    <td>

			<button
			    class="btn btn-primary btn-xs btn-view"
			    data-vendor="${d.vendor}">

			    View

			</button>

		    </td>

		</tr>

	    `;

		});

		html += `

	    </tbody>

	</table>

	`;

		this.page.main.find(".supplier-table").html(html);

	}

	bind_events(){

		const me = this;
		$(document).on("click", ".remove-cart-item", function () {

			const name = $(this).data("name");

			frappe.confirm(

				"Remove this product from cart?",

				function () {

					frappe.call({

						method: "vgjewellry.vg_jewellery.page.vg_quotation_cart.vg_quotation_cart.remove_cart_item",

						args: {
							name: name
						},

						callback(r) {

							if (r.message) {

								frappe.show_alert({
									message: "Product removed",
									indicator: "green"
								});

								me.open_vendor_cart(me.current_vendor);


								me.load_cart();

								if (me.load_cart_count) {
									me.load_cart_count();
								}

							}

						}

					});

				}

			);

		});

		$(document).on("click",".btn-view",function(){

			let vendor = $(this).data("vendor");

			me.open_vendor_cart(vendor);


		});
		$(document).on("click", "#generate-po", () => {

			let remarks = {};

			$(".cart-remark").each(function () {

				remarks[$(this).data("name")] = $(this).val();

			});

			frappe.call({

				method: "vgjewellry.vg_jewellery.page.vg_quotation_cart.vg_quotation_cart.generate_po",

				args: {

					vendor: this.current_vendor,

					vendor_delivery_date: $("#vendor_delivery_date").val(),

					remarks: remarks

				},

				callback: (r) => {

					if (!r.message) return;

					frappe.show_alert({
						message: "PO Created Successfully " ,
						indicator: "green"
					});

					this.dialog.hide();

					setTimeout(() => { location.reload(); }, 500);

				}

			});

		});
	}
	open_vendor_cart(vendor){
		this.current_vendor = vendor;
		if (!this.dialog) {

			this.dialog = new frappe.ui.Dialog({

				title: "Vendor Cart",

				size: "extra-large",

				fields:[
					{
						fieldtype:"HTML",
						fieldname:"cart"
					}
				]

			});
		}

		this.dialog.show();

		frappe.call({

			method:
			"vgjewellry.vg_jewellery.page.vg_quotation_cart.vg_quotation_cart.get_vendor_cart",

			args:{
				vendor:vendor
			},

			callback:(r)=>{
				this.dialog.hide();

				this.render_vendor_cart(r.message);

			}

		});

	}


	render_vendor_cart(data) {

		let total_net_wt = 0;
		let total_dia_wt = 0;
		let total_stone_wt = 0;

		let html = `

<div class="container-fluid">

    <div class="row mb-3">

	<div class="col-md-4">

	    <label><b>Vendor</b></label>

	    <div style="font-size:16px;font-weight:bold;">
		${data.vendor || ""}
	    </div>

	</div>

	<div class="col-md-4">

	    <label><b>Vendor Delivery Date</b></label>

	    <input
		type="date"
		class="form-control"
		id="vendor_delivery_date"
		value="${data.vendor_delivery_date || ""}"
	    >

	</div>

    </div>

`;


		/*
		 * =========================================================
		 * GROUP BY BRANCH
		 * =========================================================
		 */

		(data.branches || []).forEach((branch_data, branch_index) => {

			let branch_net_wt = 0;
			let branch_dia_wt = 0;
			let branch_stone_wt = 0;


			html += `

	<div class="card mb-4">

	    <!-- BRANCH HEADER -->

	    <div class="card-header"
		style="
		    background:#f8f9fa;
		    border:1px solid #ddd;
		    padding:12px 15px;
		    font-size:17px;
		    font-weight:600;
		">

		<i class="fa fa-building"></i>

		Branch :

		<span style="color:#0d6efd;">
		    ${branch_data.branch_name || "Not Assigned"}
		</span>

	    </div>


	    <div class="card-body p-0">

		<table class="table table-bordered table-striped"
		       style="margin:0;">

		    <thead>

			<tr>

			    <th width="50">Sr.</th>

			    <th width="120">Image</th>

			    <th>Item</th>

			    <th width="90">Metal</th>

			    <th width="90">Net Wt</th>

			    <th width="90">Gr Wt</th>

			    <th width="80">Qty</th>

			    <th width="250">Remark</th>

			    <th width="60"></th>

			</tr>

		    </thead>

		    <tbody>

	`;


			/*
			 * =====================================================
			 * ITEMS
			 * =====================================================
			 */

			(branch_data.items || []).forEach((d, i) => {

				let item_net_wt = Number(d.net_wt || 0);

				branch_net_wt += item_net_wt;
				total_net_wt += item_net_wt;


				/*
				 * =================================================
				 * DIAMOND DETAILS
				 * =================================================
				 */

				let diamond_html = "";

				let item_dia_wt = 0;


				if (
					d.diamond_details &&
					d.diamond_details.length
				) {

					diamond_html = `

		    <table class="table table-bordered table-condensed"
			   style="margin:0; background:white;">

			<thead>

			    <tr>

				<th>Shape</th>
				<th>Size</th>
				<th>Pcs</th>
				<th>Wt</th>
				<th>Rate</th>
				<th>Amount</th>

			    </tr>

			</thead>

			<tbody>
		`;


					d.diamond_details.forEach(dia => {

						let dia_wt =
							Number(dia.diamond_wt || 0);

						item_dia_wt += dia_wt;


						diamond_html += `

			<tr>

			    <td>
				${dia.diamond_shape || "-"}
			    </td>

			    <td>
				${dia.diamond_size || "-"}
			    </td>

			    <td>
				${dia.diamond_pcs || 0}
			    </td>

			    <td>
				${dia_wt.toFixed(3)}
			    </td>

			    <td>
				${Number(
					dia.diamond_rate || 0
				).toFixed(2)}
			    </td>

			    <td>
				${Number(
					dia.diamond_amount || 0
				).toFixed(2)}
			    </td>

			</tr>

		    `;

					});


					diamond_html += `

			</tbody>

		    </table>

		`;

				} else {

					diamond_html = `
		    <span class="text-muted">
			No Diamond
		    </span>
		`;

				}


				branch_dia_wt += item_dia_wt;
				total_dia_wt += item_dia_wt;


				/*
				 * =================================================
				 * STONE DETAILS
				 * =================================================
				 */
				
				let stone_html = "";

				let item_stone_wt = 0;


				if (
					d.stone_details &&
					d.stone_details.length
				) {

					stone_html = `

		    <table class="table table-bordered table-condensed"
			   style="margin:0; background:white;">

			<thead>

			    <tr>

				<th>Pcs</th>
				<th>Wt</th>
				<th>Rate</th>
				<th>Amount</th>

			    </tr>

			</thead>

			<tbody>
		`;


					d.stone_details.forEach(dia => {

						let dia_wt =
							Number(dia.stone_wt || 0);

						item_stone_wt += dia_wt;


						stone_html += `

			<tr>


			    <td>
				${dia.stone_pcs || 0}
			    </td>

			    <td>
				${dia_wt.toFixed(3)}
			    </td>

			    <td>
				${Number(
					dia.stone_rate || 0
				).toFixed(2)}
			    </td>

			    <td>
				${Number(
					dia.stone_amount || 0
				).toFixed(2)}
			    </td>

			</tr>

		    `;

					});


					stone_html += `

			</tbody>

		    </table>

		`;

				} else {

					stone_html = `
		    <span class="text-muted">
			No Stone
		    </span>
		`;

				}


				branch_stone_wt += item_stone_wt;
				total_stone_wt += item_stone_wt;





				/*
				 * =================================================
				 * MAIN PRODUCT ROW
				 * =================================================
				 */

				html += `

		<tr>

		    <td>
			${i + 1}
		    </td>


		    <td>

			<img
			    src="${
				    d.image ||
					    '/assets/frappe/images/ui-states/default-avatar.png'
			    }"
			    style="
				width:90px;
				height:90px;
				object-fit:contain;
			    "
			>

		    </td>


		    <td>

			<b>
			    ${d.item || ""}
			</b>

			<br>

			<span class="text-muted">
			    ${d.vendor_design_number || ""}
			</span>

		    </td>


		    <td>
			${d.metal || ""}
		    </td>


		    <td>
			${item_net_wt.toFixed(3)}
		    </td>


		    <td>
			${Number(
				d.gr_wt || 0
			).toFixed(3)}
		    </td>


		    <td>

			<input name="c1"
			    type="number"
			    min="1"
			    value="${d.qty || 1}"
			    class="form-control cart-qty"
			    data-name="${d.cart_name}"
			>

		    </td>


		    <td>

			<textarea name="c3"
			    rows="2"
			    class="form-control cart-remark"
			    data-name="${d.cart_name}"
			    placeholder="Enter Remark"
			>${d.remark || ""}</textarea>

		    </td>


		    <td>

			<button
			    class="btn btn-danger btn-sm remove-cart-item"
			    data-name="${d.cart_name}"
			>

			    <i class="fa fa-trash"></i>

			</button>

		    </td>

		</tr>


		<!-- =================================================
		     SECOND ROW : DIAMOND + STONE
		     ================================================= -->

		<tr style="background:#fafafa;">

		    <td></td>

		    <td></td>

		    <td colspan="2">

			<div style="
			    font-weight:600;
			    margin-bottom:5px;
			    color:#495057;
			">

			    <i class="fa fa-diamond"></i>

			    Diamond Details

			</div>

			${diamond_html}

		    </td>


		    <td colspan="3">

			<div style="
			    font-weight:600;
			    margin-bottom:5px;
			    color:#495057;
			">

			    <i class="fa fa-circle"></i>

			    Stone Details

			</div>

			${stone_html}

		    </td>


		    <td colspan="2">

			<div style="
			    font-size:13px;
			    color:#6c757d;
			    padding:10px;
			">

			    <b>Diamond Wt:</b>
			    ${item_dia_wt.toFixed(3)}

			    <br>

			    <b>Stone Wt:</b>
			    ${item_stone_wt.toFixed(3)}

			</div>

		    </td>

		</tr>

	    `;

			});


			/*
			 * =====================================================
			 * BRANCH TOTAL
			 * =====================================================
			 */

			html += `

		    </tbody>

		</table>


		<div
		    style="
			padding:12px 15px;
			background:#f8f9fa;
			border-top:1px solid #ddd;
		    "
		>

		    <div style="
			display:flex;
			gap:25px;
			align-items:center;
			font-weight:600;
			flex-wrap:wrap;
		    ">

			<div>

			    Branch Net Wt :

			    <span style="color:#0d6efd;">
				${branch_net_wt.toFixed(3)}
			    </span>

			</div>


			<div>

			    Branch Diamond Wt :

			    <span style="color:#198754;">
				${branch_dia_wt.toFixed(3)}
			    </span>

			</div>


			<div>

			    Branch Stone Wt :

			    <span style="color:#fd7e14;">
				${branch_stone_wt.toFixed(3)}
			    </span>

			</div>

		    </div>

		</div>


	    </div>

	</div>

	`;

		});


		/*
		 * =========================================================
		 * OVERALL TOTAL
		 * =========================================================
		 */

		html += `

	<div class="row mt-3 align-items-center">

	    <div class="col-md-8">

		<div style="
		    display:flex;
		    gap:15px;
		    align-items:center;
		    font-size:17px;
		    font-weight:600;
		    flex-wrap:wrap;
		">


		    <div style="
			background:#f8f9fa;
			padding:12px 18px;
			border:1px solid #ddd;
			border-radius:6px;
		    ">

			Total Net Wt :

			<span style="color:#0d6efd;">
			    ${total_net_wt.toFixed(3)}
			</span>

		    </div>


		    <div style="
			background:#f8f9fa;
			padding:12px 18px;
			border:1px solid #ddd;
			border-radius:6px;
		    ">

			Total Diamond Wt :

			<span style="color:#198754;">
			    ${total_dia_wt.toFixed(3)}
			</span>

		    </div>


		    <div style="
			background:#f8f9fa;
			padding:12px 18px;
			border:1px solid #ddd;
			border-radius:6px;
		    ">

			Total Stone Wt :

			<span style="color:#fd7e14;">
			    ${total_stone_wt.toFixed(3)}
			</span>

		    </div>


		</div>

	    </div>


	    <div class="col-md-4 text-right">

		<button
		    class="btn btn-success btn-lg"
		    id="generate-po"
		>

		    <i class="fa fa-check"></i>

		    Generate PO

		</button>

	    </div>

	</div>

    </div>

`;


		this.dialog.fields_dict.cart.$wrapper.html(html);
	}


}

File: vgjewellry/vg_jewellery/page/vg_quotation_cart/vg_quotation_cart.py
import frappe
import json
from frappe import _


@frappe.whitelist()
def get_supplier_list():

    return frappe.db.sql("""
        SELECT
            qc.vendor,
            qu.vendor_code,
            COUNT(*) AS qty
        FROM `tabQuotation Cart` qc
        INNER JOIN `tabQuotation Upload New` qu
            ON qc.quotation_id = qu.name
        WHERE qc.status = 1
        GROUP BY
            qc.vendor,
            qu.vendor_code
        ORDER BY
            qc.vendor
    """, as_dict=True)


@frappe.whitelist()
def get_vendor_cart(vendor):

    # ---------------------------------------------------------
    # GET CART + QUOTATION + BRANCH NAME
    # ---------------------------------------------------------

    items = frappe.db.sql("""
        SELECT
            qc.name AS cart_name,
            qu.*,

            qci.branch,
            obm.branch_name,

            qci.remark

        FROM `tabQuotation Cart` qc

        INNER JOIN `tabQuotation Upload New` qu
            ON qc.quotation_id = qu.name

        LEFT JOIN `tabQuotation Cart Item` qci
            ON qci.item = qu.name
            AND qci.quotation_number = qc.name

        LEFT JOIN `tabOrnate_Branch_Master` obm
            ON obm.name = qci.branch

        WHERE qc.vendor = %s
          AND qc.status = 1

        ORDER BY qu.name
    """, (vendor,), as_dict=True)


    if not items:
        return {
            "vendor": vendor,
            "vendor_code": "",
            "branches": []
        }


    # ---------------------------------------------------------
    # GET DIAMOND DETAILS
    # ---------------------------------------------------------

    quotation_names = list(set(
        [str(d.name) for d in items]
    ))


    diamond_rows = frappe.db.sql("""
        SELECT
            quotation_number,
            diamond_shape,
            diamond_size,
            diamond_pcs,
            diamond_wt,
            diamond_rate,
            diamond_amount

        FROM `tabQuotation Upload Diamond`

        WHERE quotation_number IN %(quotation_names)s

        ORDER BY
            quotation_number,
            diamond_shape,
            diamond_size
    """, {
        "quotation_names": quotation_names
    }, as_dict=True)


    # ---------------------------------------------------------
    # GROUP DIAMONDS BY QUOTATION
    # ---------------------------------------------------------

    diamond_map = {}

    for diamond in diamond_rows:

        quotation_number = str(
            diamond.quotation_number
        )

        if quotation_number not in diamond_map:
            diamond_map[quotation_number] = []


        diamond_map[quotation_number].append({

            "diamond_shape":
                diamond.diamond_shape or "",

            "diamond_size":
                diamond.diamond_size or "",

            "diamond_pcs":
                diamond.diamond_pcs or 0,

            "diamond_wt":
                diamond.diamond_wt or 0,

            "diamond_rate":
                diamond.diamond_rate or 0,

            "diamond_amount":
                diamond.diamond_amount or 0

        })


    # ---------------------------------------------------------
    # ADD DIAMOND DETAILS
    # ---------------------------------------------------------

    for item in items:

        item["diamond_details"] = diamond_map.get(
            str(item.name),
            []
        )


    # ---------------------------------------------------------
    # GET STONE DETAILS
    # ---------------------------------------------------------

    quotation_names = list(set(
        [str(d.name) for d in items]
    ))


    stone_rows = frappe.db.sql("""
        SELECT
            quotation_number,
            stone_pcs,
            stone_wt,
            stone_rate,
            stone_amount

        FROM `tabQuotation Upload Stone`

        WHERE quotation_number IN %(quotation_names)s

        ORDER BY
            quotation_number
    """, {
        "quotation_names": quotation_names
    }, as_dict=True)


    # ---------------------------------------------------------
    # GROUP STONES BY QUOTATION
    # ---------------------------------------------------------

    stone_map = {}

    for stone in stone_rows:

        quotation_number = str(
            stone.quotation_number
        )

        if quotation_number not in stone_map:
            stone_map[quotation_number] = []


        stone_map[quotation_number].append({

            "stone_pcs":
                stone.stone_pcs or 0,

            "stone_wt":
                stone.stone_wt or 0,

            "stone_rate":
                stone.stone_rate or 0,

            "stone_amount":
                stone.stone_amount or 0

        })


    # ---------------------------------------------------------
    # ADD STONE DETAILS
    # ---------------------------------------------------------

    for item in items:

        item["stone_details"] = stone_map.get(
            str(item.name),
            []
        )
    # ---------------------------------------------------------
    # GROUP BY BRANCH
    # ---------------------------------------------------------

    branches = {}


    for item in items:

        branch_id = item.branch or ""
        branch_name = item.branch_name or "Not Assigned"

        # Use branch_id as grouping key
        if branch_id not in branches:

            branches[branch_id] = {

                "branch": branch_id,

                "branch_name": branch_name,

                "items": [],

                "total_net_wt": 0

            }


        branches[branch_id]["items"].append(item)


        branches[branch_id]["total_net_wt"] += (
            float(item.net_wt or 0)
        )


    # ---------------------------------------------------------
    # RETURN
    # ---------------------------------------------------------

    return {

        "vendor": vendor,

        "vendor_code":
            items[0].vendor_code,

        "branches":
            list(branches.values())

    }



@frappe.whitelist()
def remove_cart_item(name):

    if frappe.db.exists("Quotation Cart", name):

        frappe.delete_doc(
            "Quotation Cart",
            name,
            ignore_permissions=True
        )

    return True  


@frappe.whitelist()
def generate_po(vendor, vendor_delivery_date=None, remarks=None):

    import json

    # ---------------------------------------------------------
    # PARSE REMARKS
    # ---------------------------------------------------------

    if isinstance(remarks, str):
        remarks = json.loads(remarks)

    if not remarks:
        remarks = {}


    # ---------------------------------------------------------
    # GET ACTIVE CART ITEMS
    # ---------------------------------------------------------

    cart_items = frappe.db.sql("""
        SELECT
            qc.name AS cart_name,
            qc.quotation_id,
            qc.vendor,

            qci.branch,
            qci.remark AS cart_remark

        FROM `tabQuotation Cart` qc

        LEFT JOIN `tabQuotation Cart Item` qci
            ON qci.quotation_number = qc.name
            AND qci.item = qc.quotation_id

        WHERE qc.vendor = %s
          AND qc.status = 1

        ORDER BY
            qci.branch,
            qc.creation ASC

    """, (vendor,), as_dict=True)


    if not cart_items:
        frappe.throw(
            _("No products found in cart.")
        )


    # ---------------------------------------------------------
    # GROUP CART ITEMS BY BRANCH
    # ---------------------------------------------------------

    branch_groups = {}

    for row in cart_items:

        # Branch ID
        branch = row.branch or "Not Assigned"


        if branch not in branch_groups:

            branch_groups[branch] = []


        branch_groups[branch].append(row)


    created_pos = []


    # ---------------------------------------------------------
    # CREATE ONE PO FOR EACH BRANCH
    # ---------------------------------------------------------

    for branch, branch_items in branch_groups.items():


        # -----------------------------------------------------
        # CREATE PO HEADER
        # -----------------------------------------------------

        po = frappe.new_doc("Quotation PO")

        po.vendor = vendor

        po.branch = (
            None
            if branch == "Not Assigned"
            else branch
        )

        po.vendor_delivery_date = (
            vendor_delivery_date
        )

        po.jewellery_type = "Diamond"

        po.status = 1

        po.insert(
            ignore_permissions=True
        )


        # -----------------------------------------------------
        # CREATE PO ITEMS
        # -----------------------------------------------------

        for row in branch_items:

            po_item = frappe.new_doc(
                "Quotation PO Item"
            )

            po_item.po_no = po.name

            # Quotation Upload New
            po_item.item = row.quotation_id


            # Remark from frontend
            frontend_remark = remarks.get(
                row.cart_name,
                ""
            )


            # If frontend remark exists,
            # use it. Otherwise use cart remark.
            po_item.remark = (
                frontend_remark
                or row.cart_remark
                or ""
            )


            po_item.insert(
                ignore_permissions=True
            )


            # -------------------------------------------------
            # UPDATE CART STATUS
            # -------------------------------------------------

            frappe.db.set_value(
                "Quotation Cart",
                row.cart_name,
                {
                    "status": 2
                },
                update_modified=False
            )


        # -----------------------------------------------------
        # SAVE PO
        # -----------------------------------------------------

        created_pos.append({

            "po_no": po.name,

            "vendor": vendor,

            "branch": branch,

            "vendor_delivery_date":
                po.vendor_delivery_date,

            "jewellery_type":
                po.jewellery_type,

            "item_count":
                len(branch_items)

        })


    # ---------------------------------------------------------
    # COMMIT
    # ---------------------------------------------------------

    frappe.db.commit()


    # ---------------------------------------------------------
    # RETURN
    # ---------------------------------------------------------

    return {

        "success": True,

        "vendor": vendor,

        "vendor_delivery_date":
            vendor_delivery_date,

        "jewellery_type":
            "Diamond",

        "total_po":
            len(created_pos),

        "pos":
            created_pos,

        "message":
            _(
                "{0} PO(s) created successfully."
            ).format(
                len(created_pos)
            )

    }

@frappe.whitelist()
def generate_po_old(vendor, vendor_delivery_date=None, remarks=None):

    if isinstance(remarks, str):
        remarks = json.loads(remarks)

    if not remarks:
        remarks = {}

    cart_items = frappe.get_all(
        "Quotation Cart",
        filters={
            "vendor": vendor,
            "status": 1
        },
        fields=[
            "name",
            "quotation_id",
            "vendor"
            #"vendor_delivery_date",
            #"jewellery_type"
        ],
        order_by="creation asc"
    )

    if not cart_items:
        frappe.throw(_("No products found in cart."))

    # Create PO Header
    po = frappe.new_doc("Quotation PO")
    po.vendor = vendor
    po.vendor_delivery_date = (
        vendor_delivery_date
    )
    po.jewellery_type = (
        "Diamond"
    )
    po.status = 1
    po.insert(ignore_permissions=True)

    # Create PO Items
    for row in cart_items:

        po_item = frappe.new_doc("Quotation PO Item")
        po_item.po_no = po.name

        # Link to Quotation Upload
        po_item.item = row.quotation_id

        # If these fields exist in Quotation PO Item
        po_item.vendor_delivery_date = row.vendor_delivery_date
        po_item.jewellery_type = row.jewellery_type

        # Remark from Cart Page
        po_item.remark = remarks.get(row.name, "")

        po_item.insert(ignore_permissions=True)

        # Update Cart
        frappe.db.set_value(
            "Quotation Cart",
            row.name,
            {
                "status": 2
            },
            update_modified=False
        )

    frappe.db.commit()

    return {
        "success": True,
        "po_no": po.name,
        "vendor": po.vendor,
        "vendor_delivery_date": po.vendor_delivery_date,
        "jewellery_type": po.jewellery_type,
        "message": _("Quotation PO Created Successfully.")
    }

